codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionTestKitGenerator.scala

```diff
@@ -0,0 +1,127 @@
+
+    generatedFiles
+  }
+
+  private[codegen] def generateSourceCode(
+      entity: ModelBuilder.EventSourcedEntity,
+      service: ModelBuilder.ActionService,
+      packageName: String,
+      className: String): String = {
+    val imports = generateImports(
+      commandTypes(service.commands),
+      "",
+      otherImports = Seq(
+        "java.util.ArrayList",
+        "java.util.List",
+        "java.util.function.Function",
+        "java.util.Optional",
+        "com.akkaserverless.javasdk.action.Action",
+        "com.akkaserverless.javasdk.action.ActionCreationContext",
+        "com.akkaserverless.javasdk.testkit.impl.ActionResultImpl",
+        "com.akkaserverless.javasdk.impl.action.ActionEffectImpl",
+        "com.example.actions.CounterJournalToTopicAction",
+        "com.example.actions.CounterTopicApi",
+        "com.akkaserverless.javasdk.testkit.impl.StubActionCreationContext",
+        "com.akkaserverless.javasdk.testkit.impl.StubActionContext"))
+
+    val testKitClassName = s"${className}TestKit"
+
+    println(packageName)
+
+    s"""$managedComment
+          |package ${service.fqn.parent.javaPackage};
+          |
+          |$imports
+          |
+          |public final class $testKitClassName {
+          |
+          |  private Function<ActionCreationContext, $className> actionFactory;
+          |
+          |  private CounterJournalToTopicAction createAction() {
+          |    $className action = actionFactory.apply(new StubActionCreationContext());
+          |    action._internalSetActionContext(Optional.of(new StubActionContext()));
+          |    return action;
+          |  };
+          |
+          |  public static $testKitClassName of(Function<ActionCreationContext, $className> actionFactory) {
+          |    return new $testKitClassName(actionFactory);
+          |  }
+          |
+          |  private $testKitClassName(Function<ActionCreationContext, $className> actionFactory) {
+          |    this.actionFactory = actionFactory;
+          |  }
+          |
+          |  private <E> ActionResult<E> interpretEffects(Action.Effect<E> effect) {
+          |    return new ActionResultImpl(effect);
+          |  }
+          |
+          |  ${Format.indent(generateServices(service, entity), 2)}
+          |
+          |}
+          |""".stripMargin
+  }
+
+  def generateServices(service: ModelBuilder.ActionService, entity: ModelBuilder.EventSourcedEntity): String = {
+    require(!service.commands.isEmpty, "empty `commands` not allowed")
+
+    def selectOutput(command: ModelBuilder.Command): String =
+      if (command.outputType.name == "Empty") {
+        "Empty"
+      } else {
+        command.outputType.fullName
+      }
+
+    val domainOuterClass = entity.fqn.parent.name
+
+    service.commands
+      .map { command =>
+        s"""|public ActionResult<${selectOutput(command)}> ${lowerFirst(command.name)}(${domainOuterClass}.${command.inputType.protoName} event) {
```

Comment:
it's not an `event` here, use either `command` or the actual name of the input parameter type (with `lowerFirst`).

Reply:
ok, done. 
Question why is then in the past tense?

Reply:
What is in past tense?

Reply:
for instance  the input in `public ActionResult<com.example.actions.CounterTopicApi.Increased> increase(com.example.domain.CounterDomain.ValueIncreased valueIncreased)`